stan/math/opencl/err/check_diagonal_zeros.hpp

```diff
@@ -33,8 +33,7 @@ inline void check_diagonal_zeros(const char* function, const char* name,
   cl::Context ctx = opencl_context.context();
   try {
     int zero_on_diagonal_flag = 0;
-    matrix_cl<int> zeros_flag(1, 1);
-    zeros_flag = to_matrix_cl(zero_on_diagonal_flag);
+    matrix_cl<int> zeros_flag = constant(0, 1, 1);
```

Comment:
[optional] This looks good, but we could completely replace this function with kernel generator implementation.